### PR TITLE
[Enhancement] [zos_operator] Case sensitivity in zos_operator commands

### LINKED ISSUE #1155: [Enhancement] [zos_operator] add support for case sensitivity
### Is there an existing issue for this?

- [X] There are no existing issues.

### Ansible module

zos_operator

### Enhancement or feature description

When sending a command that has case sensitive character such as a USS path, the command and character that are case sensitive are capitalized,  this does not happen in the Ansible code but instead , the ZOAU api is not instructed to not capetalize thus requesting that we add and option to the zos_operator module that allows a user to control the case sensitivity.  

e.g.
```
        - name: Sending Start ZCPA Command
          ibm.ibm_zos_core.zos_operator:
            cmd: "S ZCPA,JOBNAME=ZCPA1,CONF='/global/zcpa_zos/instances/ZCPA1/start.json'"
          when: zcpa_status.msg == "*"
```

Results in (notice `CONF='/GLOBAL/ZCPA_ZOS/INSTANCES/ZCPA1/START.JSON`):
```
task path: /Users/kippins/Code/zcontainer-solution-test/ansible/playbooks/start-kubernetes.yaml:76
changed: [J78] => {"changed": true, "cmd": "S ZCPA,JOBNAME=ZCPA1,CONF='/global/zcpa_zos/instances/ZCPA1/start.json'", "content": ["J78        2023264  11:22:09.00             ISF031I CONSOLE KIPP0000 ACTIVATED", "J78        2023264  11:22:09.00            -S ZCPA,JOBNAME=ZCPA1,CONF='/GLOBAL/ZCPA_ZOS/INSTANCES/ZCPA1/START.JSON' ", "J78        2023264  11:22:09.00             GLZB025I zCX instance ZCPA1: Initialization has started. Code date 07/28/23.", "                                           GLZB004I z/OS UNIX Service BPX4OPN failed, RC=00000081 RSN=0594003D", "                                           PATH=/GLOBAL/ZCPA_ZOS/INSTANCES/ZCPA1/START.JSON.", "                                           GLZB002I zCX instance ZCPA1 has ended.", "                                           GLZB005I z/OS system service GLZBAIN failed, RC=00000008 RSN=0101000B", "                                           zCX ended due to failure."], "elapsed": 1.88, "rc": 0, "wait_time_s": 1}
```

Work around:
```
    - name: Execute opercmd
      command: opercmd -p "S ZCPA,JOBNAME=ZCPA1,CONF='/global/zcpa_zos/instances/ZCPA1/start.json'"
      register: result
```

I suggest we add a new option that preserves case, one of these:
- case_preserving: (bool)
- case_sensitive: (bool)
- preserve: (bool)

See API option `preserve` [here](https://www.ibm.com/docs/en/zoau/1.3.x?topic=apis-opercmd)

The other option is to change the default behavior such case sensitivity is the default and introduce an option such as (below), but to change the behavior we would want to educate inform our users and also research if this is a good approach or not. 
- case_preserving: (bool)
- case_sensitive: (bool)
- uppercase: (bool)

### PR DESCRIPTION
**Pull request by rexemin**
##### SUMMARY
Fixes #1155.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
zos_operator

##### ADDITIONAL INFORMATION
Added a new option to the module, called `case_sensitive`, which is a boolean telling the module that the command is case sensitive and therefore not change the casing. Can change the name of the option, though.

### CODE REVIEW

**Comment by rexemin** [comment]
Results for the following modules:
- /tests/functional/modules/test_zos_operator_func.py 
- /tests/functional/modules/test_zos_apf_func.py 
- /tests/functional/modules/test_zos_mvs_raw_func.py 
- /tests/functional/modules/test_zos_operator_action_query_func.py 
- /tests/unit/test_zos_operator_unit.py 
- /tests/functional/modules/test_zos_fetch_func.py 
- /tests/functional/modules/test_zos_find_func.py 
- /tests/functional/modules/test_zos_job_submit_func.py 
- /tests/functional/modules/test_zos_mount_func.py 
- /tests/functional/modules/test_zos_volume_init_func.py

Screenshot 2024-08-12 at 16 07 09
Screenshot 2024-08-12 at 16 07 20

**Comment by rexemin** [comment]
Results for
- /tests/functional/modules/test_zos_data_set_func.py

Screenshot 2024-08-12 at 17 23 41
Screenshot 2024-08-12 at 17 23 49

**Comment by rexemin** [comment]
Results for:
- /tests/functional/modules/test_zos_copy_func.py

Screenshot 2024-08-13 at 15 16 34
Screenshot 2024-08-13 at 15 16 43

**Review by ddimatos — CHANGES_REQUESTED** [review]
Looks good, 2 comments:
- Please update the copyright year on the test case, easy to overlook, makes the one releasing's job a bit easier.
- This won't yield a different result but maybe clarity, change the command:
```
command = "d u,all"
```
to
```
command = "D U,all"
```
For mixed case, should still work on z/OS. Just a suggestion.

**Review by richp405 — APPROVED** [review]

File: changelogs/fragments/1641-case-sensitivity-zos_operator.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+minor_changes:
+  - zos_operator - add new option case_sensitive to module, allowing users
```
Comment (ketankelkar) [inline changelogs/fragments/1641-case-sensitivity-zos_operator.yml:2]: nit-picking here, in the release notes (which are built from the changelog), we tend to capitalize first word, use past tense, and put some formatting around any module options mentioned
> Added a new option ``case_sensitive`` to ...

**Review by ketankelkar — APPROVED** [review]
lgtm, trying something new  - nitpicking changelog fragments now instead of during releases

File: changelogs/fragments/1641-case-sensitivity-zos_operator.yml (hunk 1 above)
Comment (rexemin) [reply changelogs/fragments/1641-case-sensitivity-zos_operator.yml:2]: Will make those changes

**Review by fernandofloresg — APPROVED** [review]
LGTM

**Review by ddimatos — APPROVED** [review]
Looks good, thanks for the handling the suggested edits.

**Pull request merged**